File: backend/src/main/java/com/learnkk/kernel/domain/MeetingStatus.java
package com.learnkk.kernel.domain;

/**
 * Meeting lifecycle status (contract #3). Physical representation is varchar + CHECK in the DB.
 *
 * <p>Bolt 1 exercises only PENDING_APPROVAL -&gt; RECRUITING (T1) and PENDING_APPROVAL -&gt;
 * REJECTED (T2). The remaining transitions are declared for the schema but wired in Bolt 2+.
 */
public enum MeetingStatus {
  PENDING_APPROVAL,
  RECRUITING,
  READY_TO_START,
  IN_PROGRESS,
  COMPLETED,
  REJECTED,
  CANCELLED
}

File: frontend/src/features/shared/meetingStatus.ts
import type { MeetingStatus } from '@/api';

const STATUS_LABELS: Record<MeetingStatus, string> = {
  PENDING_APPROVAL: '승인대기',
  RECRUITING: '모집중',
  READY_TO_START: '시작대기',
  IN_PROGRESS: '진행중',
  COMPLETED: '완료',
  REJECTED: '반려',
  CANCELLED: '취소',
};

export type BadgeVariant = 'default' | 'secondary' | 'destructive' | 'outline';

const STATUS_VARIANTS: Record<MeetingStatus, BadgeVariant> = {
  PENDING_APPROVAL: 'outline',
  RECRUITING: 'default',
  READY_TO_START: 'secondary',
  IN_PROGRESS: 'secondary',
  COMPLETED: 'secondary',
  REJECTED: 'destructive',
  CANCELLED: 'destructive',
};

export function meetingStatusLabel(status: MeetingStatus): string {
  return STATUS_LABELS[status] ?? status;
}

export function meetingStatusVariant(status: MeetingStatus): BadgeVariant {
  return STATUS_VARIANTS[status] ?? 'secondary';
}
